src/inference/text_emotion_service.py

```
@@ -0,0 +1,96 @@
+            from transformers import pipeline, AutoTokenizer, AutoModelForSequenceClassification  # type: ignore
+        except Exception as e:  # pragma: no cover
+            logger.error("Failed to import transformers components: %s", e)
+            raise
+
+        model_dir = os.environ.get(self.model_dir_env)
+        local_only = os.environ.get(self.local_only_env, "1").strip() not in {"", "0", "false", "False"}
+
+        if not model_dir and local_only:
+            model_dir = DEFAULT_LOCAL_MODEL_DIR
+
+        if model_dir and os.path.isdir(model_dir):
```

Ruff (0.12.2)

`os.path.isdir()` should be replaced by `Path.is_dir()`

(PTH112)